virtcontainers/device/drivers/block.go

```diff
@@ -88,53 +78,68 @@ func (device *BlockDevice) Attach(devReceiver api.DeviceReceiver) (err error) {
 		return err
 	}
 
-	drive := Drive{
-		File:   device.DeviceInfo.HostPath,
-		Format: "raw",
-		ID:     utils.MakeNameID("drive", device.DeviceInfo.ID, maxDevIDSize),
-		Index:  index,
-	}
-
-	deviceLogger().WithField("device", device.DeviceInfo.HostPath).Info("Attaching block device")
-	device.BlockDrive = &drive
-	if err = devReceiver.HotplugAddDevice(device, config.DeviceBlock); err != nil {
-		return err
+	drive := &config.BlockDrive{
+		File:          device.DeviceInfo.HostPath,
+		Format:        "raw",
+		ID:            utils.MakeNameID("drive", device.DeviceInfo.ID, maxDevIDSize),
+		Index:         index,
+		ContainerPath: device.DeviceInfo.ContainerPath,
 	}
 
-	device.DeviceInfo.Hotplugged = true
-
 	driveName, err := utils.GetVirtDriveName(index)
 	if err != nil {
 		return err
 	}
 
 	customOptions := device.DeviceInfo.DriverOptions
 	if customOptions != nil && customOptions["block-driver"] == "virtio-blk" {
-		device.VirtPath = filepath.Join("/dev", driveName)
-		device.PCIAddr = drive.PCIAddr
```

Comment:
Why  is the PCIAddr assignment deleted here? kata-agent relies on the PCIAddr for determining the virtio-block device name inside the guest. The assignment to Virtpath is not used by kata-agent, but was retained for the clear-container agent until that becomes obsolete.

Reply:
It's not removed, just moved to struct `BlockDrive`:

 https://github.com/kata-containers/runtime/pull/301/files#diff-50f996b3aaf4e1ec24964719e24d75c5R115